pandas/core/internals/construction.py

```diff
@@ -242,7 +242,10 @@ def init_dict(data, index, columns, dtype=None):
 
         # no obvious "empty" int column
         if missing.any() and not is_integer_dtype(dtype):
-            if dtype is None or np.issubdtype(dtype, np.flexible):
+            if is_dtype_equal(dtype, "string"):
```

Comment:
this needs to be much more general, use `is_extension_dtype``

Reply:
Is it `is_extension_array_dtype` method ?